tests/test_eval.py

```diff
@@ -0,0 +1,22 @@
+
+
+def test_eval(capture):
+    from pybind11_tests import example_eval
+
+    with capture:
```

Comment:
For this test, I think all testing logic should be moved to Python and capture/std::cout eradicated; instead of doing the actual testing, the cpp file could just expose a few functions like

```cpp
.def("eval_single_statement", 
    [&](std::string s) { py::eval<py::eval_single_statement>(s, main_namespace); })
.def("eval_expr",
    [&](std::string s) { return (py::object) py::eval(s, main_namespace); })
.def("eval_file",
    [&](std::string s) { return (py::object) py::eval_file(s, main_namespace); })
```

so they can be called from Python, making use of pytest assertions etc.